packages/math-input/src/components/icon.tsx

```diff
@@ -3,30 +3,28 @@
  */
 
 import {StyleSheet} from "aphrodite";
-import PropTypes from "prop-types";
 import * as React from "react";
 
 import {IconTypes} from "../consts";
 
 import {offBlack} from "./common-style";
 import MathIcon from "./math-icon";
-import {iconPropType} from "./prop-types";
 import SvgIcon from "./svg-icon";
 import TextIcon from "./text-icon";
 
+import type {Icon as IconPropType} from "../types";
+import type {CSSProperties} from "aphrodite";
+
 const focusedColor = "#FFF";
 const unfocusedColor = offBlack;
 
-class Icon extends React.PureComponent {
-    static propTypes = {
-        focused: PropTypes.bool,
-        icon: iconPropType.isRequired,
-        // An Aphrodite style object, or an array of Aphrodite style objects.
-        // Note that custom styles will only be applied to text and math icons
-        // (and not SVG icons).
-        style: PropTypes.any,
-    };
+type Props = {
+    focused: boolean;
```

Comment:
I could be wrong in my line of thinking, so feel free to call me out on what I'm about to say:

- Optional parameters add complexity (because you have a null value to check for)
- They also hide dead code branches (because they can create whole paths that aren't ever used)

My understanding is that we have two main exports in MathInput: the provided keypad and the input. So most of these components are internal to the package (even though we do export everything).

So what I did in this series of PRs was:

- Make everything required
- Use the errors to figure out what we actually considered optional

This helped uncover possible dead branches (which I tried making tickets for) and by the end of it, a lot of this became either optional (because that was correct) or more strongly typed. For instance with `Icon`, everything that used it used it with `focused`; so, at least to my thinking, why make it optional when that just adds needless complexity?

Since it's just types, I don't think this adds any problems for learners, but will hopefully give us more insight on how our code works.

Reply:
That reasoning makes sense.  If the types check, then I agree with making props required!